pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -553,8 +546,6 @@ def read(self, length=None, *args, **kwargs):
 
         self.bytes_read += len(chunk)
         self.limit -= len(chunk)
-        if len(chunk) == 0:
-            self.hit_eof = True
         return chunk
 
     def readline(self, size=None, *args, **kwargs):
```

Comment:
Off topic: This doesn't always respect the readline contract. In particular, if `self._peeked_data == '\n'` then blank lines can cause trouble. For example
```
>>> from pfs_middleware.middleware import LimitedInput
>>> fp = open('tox.ini')
>>> li = LimitedInput(fp, 1000)
>>> li.readline(), li.has_more_to_read
('[tox]\n', True)
>>> li.readline(), li.has_more_to_read
('envlist = py35,py27,pep8\n', True)
>>> li.readline(), li.has_more_to_read
('minversion = 1.6\n', True)
>>> li.readline(), li.has_more_to_read
('skipsdist = True\n', True)
>>> li.readline(), li.has_more_to_read
('\n[testenv]\n', True)  # has two lines!
```